Implemented Docker and Docker Compose support (#63)
TheLartians/ModernCppStarter: README.md

cmake --build build --target GenerateDocs
```

### Build and Run Docker image

The project has support to Docker, that allows building and running tests inside a container.
This container builds all targets, so you can run any of them passing a command after "sh -c".

```bash
docker build . -t modern-cpp-starter

docker run modern-cpp-starter
```

Running tests

```bash
docker build . -t modern-cpp-starter

docker run modern-cpp-starter sh -c "./build/test/GreeterTests"
```

### Build and Run Docker Compose

The project has support to Docker Compose, that allows building and running your project with other containers.

> This container builds all targets, so you can run any of them passing a command after "sh -c".

Reviewer: Also perhaps it makes sense to move this into a separate `##Docker` section below `##Usage`  as it is unrelated to library development itself.

> This container builds all targets, so you can run any of them passing a command after "sh -c".

Author: Let me know if more changes are needed!

> This container builds all targets, so you can run any of them passing a command after "sh -c".

Reviewer: Thanks for the update! I'm still wondering if we require both sections as they are very similar. Maybe only giving instructions for `docker-compose` would already be sufficient?

> This container builds all targets, so you can run any of them passing a command after "sh -c".

Author: I think it's important to have both `docker` and `docker-compose` instructions. But we can keep them in the same section. 

For example, if someone is running the modern-cpp and a Mongo container from the `docker-compose`, it's important to know how to execute the just modern-cpp container.

Ccache can be enabled by configuring with `-DUSE_CCACHE=<ON | OFF>`.

> ## Docker
> 
> The project has support to Docker, that allows building and running tests inside a container.
> This container builds all targets, so you can run any of them passing a command after `sh -c`.
> 
> ```bash
> docker build . -t modern-cpp-starter
> 
> docker run modern-cpp-starter
> ```
> 
> Running tests.
> 
> ```bash
> docker build . -t modern-cpp-starter
> 
> docker run modern-cpp-starter sh -c "./build/test/GreeterTests"
> ```
> 
> The docker-compose can be used to manage a multi-container applications. Since this container builds all targets, you can build any specific targer passing its name after "sh -c".
> 
> Running all containers.
> 
> ```bash
> docker-compose up --build
> ```
> 
> Running only your project.
> 
> ```bash
> docker-compose run application
> ```
> 
> Runing tests.
> 
> ```bash
> docker-compose run application sh -c "./build/test/GreeterTests"
> ```
> 

Reviewer: How about bundling the commands and removing the text duplication in the docs?

``````suggestion
## Docker

The project supports Docker out-of-the-box, which allows building and running the project inside a container.
The container builds all targets, so you can run any of them passing a command after `sh -c`.

### Docker Workflow

```bash
# build
docker build . -t modern-cpp-starter
# run tests
docker run modern-cpp-starter sh -c "./build/test/GreeterTests"
# run standalone
docker run modern-cpp-starter
```

### Docker Compose Workflow

```bash
# build
docker-compose up --build
# run tests
docker-compose run application sh -c "./build/test/GreeterTests"
# run standalone
docker-compose run application
```

``````
